utils/s3/client.cc

```diff
@@ -280,7 +282,7 @@ storage_io_error map_s3_client_exception(std::exception_ptr ex) {
     } catch (const httpd::unexpected_status_error& e) {
         auto status = e.status();
 
-        if (is_redirect_status(status) || status == http::reply::status_type::not_found) {
+        if (http::reply::classify_status(status) == http::reply::status_class::redirection || status == http::reply::status_type::not_found) {
```

Comment:
This (and removal of `is_redirect_status` above) doesn't fit the "Move retryable functionality out of s3 client " title